### ISSUE TITLE
No devices found

### ISSUE THREAD

**Issue by vltnm** [p0l0/hagruenbeck_cloud#117]
## System Information

version | core-2024.8.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.46-haos
arch | x86_64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 4994
Installed Version | 2.0.1
Stage | running
Available Repositories | 1468
Downloaded Repositories | 30

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.08.0
agent_version | 1.6.0
docker_version | 26.1.4
disk_total | 30.8 GB
disk_used | 26.1 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Z-Wave JS (0.6.2), Samba share (12.3.2), Terminal & SSH (9.14.0), Grafana (10.0.0), InfluxDB (5.0.0), File editor (5.8.0), ESPHome (2024.8.3), MQTT Explorer (browser-1.0.1), evcc (0.130.7), Whisper (2.1.2), Piper (1.5.2), openWakeWord (1.10.0), Music Assistant Server (2.2.2)

</details>

<details><summary>Dashboards</summary>

dashboards | 17
-- | --
resources | 19
views | 19
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 11. August 2024 um 09:56
-- | --
current_recorder_run | 5. September 2024 um 17:05
estimated_db_size | 2187.41 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

<details><summary>Solcast PV Forecast</summary>

can_reach_server | pending
-- | --
used_requests | 1
rooftop_site_count | 1

</details>

<details><summary>Bose SoundTouch Plus</summary>

integration_version | v1.0.61
-- | --
devices_configured | 2: Küche (SoundTouch 10), Büro (SoundTouch 10)

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/p0l0/hagruenbeck_cloud/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

Got new Grünbeck SE21. Connected it to the cloud and it is possible manage it via app.
Integration does not find any devices in my account.

### Reproduction steps

1. Add integration
2. fill in account details
3. click procced

### Debug logs

```text
2024-09-06 08:16:05.886 DEBUG (MainThread) [custom_components.gruenbeck_cloud.config_flow] Got 0 Devices
```

**Comment by moerpel** [comment]
Same for me having an SE18. 
** 2024-09-25 08:34:17.256 DEBUG (MainThread) [custom_components.gruenbeck_cloud.config_flow] Got 0 Devices**
The device is available in the app.

**Comment by p0l0** [comment]
I'm still investigating this, it's not fixed yet.

**Comment by p0l0** [comment]
@vltnm or @moerpel do you have the possibility to execute the [example.py](https://github.com/p0l0/pygruenbeck_cloud/blob/main/example.py) from the pygruenbeck_cloud library, and send me the response for the  https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices?api-version=2020-08-03 request?

Before executing, you need to put you login data in line [30](https://github.com/p0l0/pygruenbeck_cloud/blob/main/example.py#L30) and [31](https://github.com/p0l0/pygruenbeck_cloud/blob/main/example.py#L31) and install the pygruenbeck_cloud library (```pip install pygruenbeck_cloud```)

It should look like this:

```
[03/Oct/2024 19:28:47] DEBUG [__main__._http_request:929] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices?api-version=2020-08-03 with status 200 was [{'type': 18, 'hasError': False, 'id': 'softliQ.D/ED14672572', 'series': 'softliQ.D', 'serialNumber': 'ED14672572', 'name': 'softIQ:SD18', 'register': True}]
```

**Comment by vltnm** [comment]
`[04/Oct/2024 09:16:37] DEBUG [__main__._http_request:915] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices?api-version=2020-08-03 with method GET
[04/Oct/2024 09:16:40] DEBUG [__main__._http_request:929] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices?api-version=2020-08-03 with status 200 was []
[04/Oct/2024 09:16:40] WARNING [__main__.init:37] No devices found!
`

**Comment by p0l0** [comment]
@vltnm  can you try the version from the _[next](https://github.com/p0l0/pygruenbeck_cloud/tree/next)_ branch? Seems like Grünbeck has a new api-version, where SE Models should be supported.

You must install the package version from the branch, so the install command for the package is ```pip install https://github.com/p0l0/pygruenbeck_cloud/tree/next```

**Comment by vltnm** [comment]
It's working

**Comment by p0l0** [comment]
Great! Can you test version 1.0.0b2? It has the newer library version

**Comment by moerpel** [comment]
Yeah, device is found. Although most of the entities are unavailable. Not sure whether these will be available once they have changed.

**Comment by vltnm** [comment]
Device found 👍🏻
but all sensors are unknown

**Comment by moerpel** [comment]
Some more info. Seems like the api delivers the data but the integration does not crunch it properly. 
Sensitive data like serial number or access token was shortened or deleted. 


`2024-10-07 08:45:47.882 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Regularly updated gruenbeck_cloud data
2024-10-07 08:45:47.883 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/?api-version=2024-05-02 with method GET
2024-10-07 08:45:48.170 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/?api-version=2024-05-02 with status 200 was {'errors': [], 'timeZone': '+00:00', 'regMode': 0, 'estatus': 0, 'regTrig': '0', 'startUp': '2024.09.24', 'swVersion': '0004.0010', 'hwVersionCl': '00000006', 'sncu': 'AH2407170150', 'calcRegMo1': '02:00', 'installedOn': '2024-08-14T06:13:52.7202012', 'regMo1': '2024-10-07T02:00:00', 'type': 118, 'hasError': False, 'isRegistered': True, 'id': 'softliQ.SE/BS110', 'series': 'softliQ.SE', 'serialNumber': 'BS110', 'name': 'Weichei', 'register': False}
2024-10-07 08:45:48.173 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/parameters?api-version=2024-05-02 with method GET
2024-10-07 08:45:48.221 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS11/parameters?api-version=2024-05-02 with status 200 was {'$pending': [], 'pallowemail': True, 'pallowpushnotification': True, 'phdist': 24, 'pmaxvolmaxcap': 1.46, 'psetsoft': 6.0, 'pminvolmaxcap': 0.54, 'pwashingout': 4.0, 'pminvolmincap': 0.125, 'pregtu3': '--:--', 'pcurrent': 700, 'pregmo2': '--:--', 'pregwe1': '--:--', 'planguage': 1, 'phcurrent': 300, 'psetcapmo': 20.0, 'ptimezone': '02:00', 'pmaxvolmincap': 0.35, 'pregfr1': '--:--', 'pmaintint': 365, 'pregmo1': '02:00', 'pfreqregvalve': 140, 'pregwe2': '--:--', 'poverload': False, 'phload': 400, 'pdate': '2024.10.07', 'pforcedregdist': 3, 'pmonregmeter': 20.0, 'pmonsalting': 75.0, 'pregtu2': '--:--', 'pregwe3': '--:--', 'pregsa1': '--:--', 'pbuzzer': False, 'ppratesoftwater': 0.0043, 'pregsu1': '--:--', 'pregth1': '--:--', 'pregtu1': '--:--', 'pinspect': 180, 'pmondisinf': True, 'pdstdby': True, 'pmode': 2, 'pregmode': False, 'pclearcntreg': False, 'purlcloud1': 'gruenbeckeu01.azure-devices.net', 'phunit': False, 'pregsa2': '--:--', 'pled': 8, 'pdlstauto': True, 'prawhard': 17.0, 'purlcloud2': '', 'phwashingout': 5.0, 'phbackwash': 12.0, 'pregsa3': '--:--', 'pregsu3': '--:--', 'pactcurrent': 3, 'pfreqblendvalve': 300, 'pnomflow': 1.8, 'pregsu2': '--:--', 'pmonblend': False, 'pregfr3': '--:--', 'pregmo3': '--:--', 'pbackwash': 9.0, 'purlcert': 'prodeugruenbeckfirmware.blob.core.windows.net/cert', 'pdurcurrent': 600, 'pledbright': 80, 'pload': 1400, 'pregfr2': '--:--', 'pmodedesinf': 1, 'pntpsync': True, 'pprateregwater': 0.0039, 'pfreqregvalve2': 200, 'pprateblending': 0.004, 'pclearerrmem': False, 'pclearcntwater': False, 'pcontinent': 0, 'pmaxdurdisinfect': 4.0, 'pregth3': '--:--', 'prinsing': 9.5, 'pvolhyg': 6.8, 'pregth2': '--:--'}
2024-10-07 08:45:48.223 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/realtime/negotiate with method GET
2024-10-07 08:45:48.225 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Finished fetching gruenbeck_cloud data in 0.342 seconds (success: True)
2024-10-07 08:45:48.225 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 258, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 453, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 168, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 533, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
    self.__async_calculate_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1071, in __async_calculate_state
    if extra_state_attributes := self.extra_state_attributes:
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/gruenbeck_cloud/sensor.py", line 390, in extra_state_attributes
    return self.entity_description.extra_attr_fn(self.coordinator.data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/gruenbeck_cloud/sensor.py", line 97, in 
    "daily_usage": [daily.to_dict() for daily in device.water]  # type: ignore[union-attr]  # noqa: E501
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable
2024-10-07 08:45:48.290 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/realtime/negotiate with status 200 was {'url': 'https://prod-eu-gruenbeck-signalr.service.signalr.net/client/?hub=gruenbeck', 'accessToken': 'deleted'}
2024-10-07 08:45:48.290 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-signalr.service.signalr.net/client/negotiate?hub=gruenbeck with method POST
2024-10-07 08:45:48.412 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-signalr.service.signalr.net/client/negotiate?hub=gruenbeck with status 200 was {'negotiateVersion': 0, 'connectionId': '_vlaWNk-t4Xjs', 'availableTransports': [{'transport': 'WebSockets', 'transferFormats': ['Text', 'Binary']}, {'transport': 'ServerSentEvents', 'transferFormats': ['Text']}, {'transport': 'LongPolling', 'transferFormats': ['Text', 'Binary']}]}
2024-10-07 08:45:48.541 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/realtime/enter?api-version=2024-05-02 with method POST
2024-10-07 08:45:48.690 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/realtime/enter?api-version=2024-05-02 with status 200 was {}
2024-10-07 08:45:48.690 ERROR (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response status code for https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS110/realtime/enter?api-version=2024-05-02 is 200, we expected one of [<HTTPStatus.ACCEPTED: 202>].`

**Comment by p0l0** [comment]
Thanks @moerpel. The response object for SE Devices seems to be a little bit different to the old ones.

I will check it and prepare a new version as soon as possible.

**Comment by p0l0** [comment]
Can you please test if you now get data with version 1.0.0b3?

**Comment by moerpel** [comment]
Tested. Unfortunately still unknown. 
![Uploading IMG_0291.jpeg…]()

**Comment by moerpel** [comment]
`2024-10-18 17:22:09.562 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/serialno/realtime/enter?api-version=2024-05-02 with status 200 was {}
2024-10-18 17:22:09.562 ERROR (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response status code for https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/serialno/realtime/enter?api-version=2024-05-02 is 200, we expected one of [<HTTPStatus.ACCEPTED: 202>].
2024-10-18 17:22:10.650 ERROR (MainThread) [aiohttp_sse_client.client] fetch https://api.home-connect.com/api/homeappliances/events failed: 503 `

**Comment by ChrisDietrich** [comment]
I'm having the same issue (1.0.0b1 leads to error `Got 0 Devices`). Is there any way I can help? From the `example.py` in pygruenbeck_cloud I can see the following response
```
Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net:443/api/devices?api-version=2024-05-02 with status 200 was [{'type': 118, 'hasError': False, 'isRegistered': True, 'id': 'softliQ.SE/B????????5', 'series': 'softliQ.SE', 'serialNumber': 'B????????5', 'name': 'S????????', 'register': False}]
```
and a resulting python pretty print of a device is
```python
Device(type=118,
        has_error=False,
        id='softliQ.SE/B????????5',
        series='softliQ.SE',
        serial_number='B????????5',
        name='S????????',
        register=False,
        _next_regeneration_raw=None,
        time_zone=None,
        startup=None,
        last_service=None,
        errors=None,
        salt=None,
        water=None,
        hardware_version=None,
        mode=None,
        nominal_flow=None,
        raw_water=None,
        soft_water=None,
        software_version=None,
        unit=None,
        realtime=DeviceRealtimeInfo(soft_water_quantity=None,
                                    soft_water_quantity_2=None,
                                    regeneration_counter=None,
                                    current_flow_rate=None,
                                    current_flow_rate_2=None,
                                    remaining_capacity_volume=None,
                                    remaining_capacity_volume_2=None,
                                    remaining_capacity_percentage=None,
                                    remaining_capacity_percentage_2=None,
                                    salt_range=None,
                                    salt_consumption=None,
                                    next_service=None,
                                    regeneration_remaining_time=None,
                                    regeneration_step=None,
                                    make_up_water_volume=None,
                                    exhausted_percentage=None,
                                    actual_value_soft_water_hardness=None,
                                    capacity_figure=None,
                                    flow_rate_peak_value=None,
                                    exchanger_peak_value=None,
                                    exchanger_peak_value_2=None,
                                    last_regeneration_exchanger=None,
                                    last_regeneration_exchanger_2=None,
                                    regeneration_flow_rate_exchanger=None,
                                    regeneration_flow_rate_exchanger_2=None,
                                    blending_flow_rate=None,
                                    step_indication_regeneration_valve=None,
                                    step_indication_regeneration_valve_2=None,
                                    current_chlorine=None,
                                    remaining_amount_of_water=None),
        parameters=DeviceParameters(dlst=None,
                                    buzzer=None,
                                    buzzer_from=None,
                                    buzzer_to=None,
                                    push_notification=None,
                                    email_notification=None,
                                    water_hardness_unit=None,
                                    raw_water_hardness=None,
                                    soft_water_hardness=None,
                                    mode=None,
                                    mode_individual_monday=None,
                                    mode_individual_tuesday=None,
                                    mode_individual_wednesday=None,
                                    mode_individual_thursday=None,
                                    mode_individual_friday=None,
                                    mode_individual_saturday=None,
                                    mode_individual_sunday=None,
                                    regeneration_mode=None,
                                    regeneration_time_monday_1=None,
                                    regeneration_time_monday_2=None,
                                    regeneration_time_monday_3=None,
                                    regeneration_time_tuesday_1=None,
                                    regeneration_time_tuesday_2=None,
                                    regeneration_time_tuesday_3=None,
                                    regeneration_time_wednesday_1=None,
                                    regeneration_time_wednesday_2=None,
                                    regeneration_time_wednesday_3=None,
                                    regeneration_time_thursday_1=None,
                                    regeneration_time_thursday_2=None,
                                    regeneration_time_thursday_3=None,
                                    regeneration_time_friday_1=None,
                                    regeneration_time_friday_2=None,
                                    regeneration_time_friday_3=None,
                                    regeneration_time_saturday_1=None,
                                    regeneration_time_saturday_2=None,
                                    regeneration_time_saturday_3=None,
                                    regeneration_time_sunday_1=None,
                                    regeneration_time_sunday_2=None,
                                    regeneration_time_sunday_3=None,
                                    maintenance_interval=None,
                                    installer_name=None,
                                    installer_phone=None,
                                    installer_email=None,
                                    ntp_sync=None,
                                    fault_signal_contact=None,
                                    knx=None,
                                    nominal_flow_monitoring=None,
                                    disinfection_monitoring=None,
                                    led_ring_mode=None,
                                    led_ring_flash_on_signal=None,
                                    led_ring_brightness=None,
                                    residual_capacity_limit=None,
                                    current_setpoint=None,
                                    charge=None,
                                    interval_forced_regeneration=None,
                                    end_frequency_regeneration_valve=None,
                                    end_frequency_regeneration_valve_2=None,
                                    end_frequency_blending_valve=None,
                                    treatment_volume=None,
                                    soft_water_meter_pulse_rate=None,
                                    blending_water_meter_pulse_rate=None,
                                    regeneration_water_meter_pulse_rate=None,
                                    capacity_figure_monday=None,
                                    capacity_figure_tuesday=None,
                                    capacity_figure_wednesday=None,
                                    capacity_figure_thursday=None,
                                    capacity_figure_friday=None,
                                    capacity_figure_saturday=None,
                                    capacity_figure_sunday=None,
                                    nominal_flow_rate=None,
                                    regeneration_monitoring_time=None,
                                    salting_monitoring_time=None,
                                    slow_rinse=None,
                                    backwash=None,
                                    washing_out=None,
                                    minimum_filling_volume_smallest_cap=None,
                                    maximum_filling_volume_smallest_cap=None,
                                    minimum_filling_volume_largest_cap=None,
                                    maximum_filling_volume_largest_cap=None,
                                    longest_switch_on_time_chlorine_cell=None,
                                    maximum_remaining_time_regeneration=None,
                                    language=None,
                                    programmable_output_function=None,
                                    programmable_input_function=None,
                                    reaction_to_power_failure=None,
                                    chlorine_cell_mode=None,
                                    blending_monitoring=None,
                                    system_overloaded=None,
                                    ppressurereg=None),
        logger=<Logger pygruenbeck_cloud.models (WARNING)>)
```

I'm running HA 2024.10.4 and installed `hagruenbeck_cloud` via HACS 2. For some reason it only shows 1.0.0b1, neither 1.0.0b2 nor 1.0.0b3.

**Comment by ChrisDietrich** [comment]
Managed to install 1.0.0b3 which now sees the SE device, however (as already mentioned by @moerpel ) the entities are in state unknown. Played with the example.py a bit, and it seems the parameters are returned correctly in the API response, but are not set in the object. Maybe somewhere in [`get_device_infos_parameters(..)`](https://github.com/p0l0/pygruenbeck_cloud/blob/next/pygruenbeck_cloud/pygruenbeck_cloud.py#L544) but I don't see anything obvious. Is there anything I can contribute to help deal with this issue?

**Comment by purche42** [comment]
Same issue with my  SE model.  Pre-release now finds my device but all sensors are unknown. How can I apply the fix or will there be a new pre-release ?

Thanks you for your work on this integration!!

**Comment by vltnm** [comment]
Hey @p0l0 ,
ist there anything we can do to support you getting SE models running?
Greetings

**Comment by HankODonald** [comment]
#

> Same issue with my SE model. Pre-release now finds my device but all sensors are unknown. How can I apply the fix or will there be a new pre-release ?
> 
> Thanks you for your work on this integration!!

Hello, I had the same error. Also, my values ​​were no longer being updated in the Android app. I contacted Grünbeck Support. They determined that the HA integration was causing too many queries to the cloud and to my softliQ:SE18. This is not currently supported and is likely causing a timeout on my softliQ:SE18.

**Comment by purche42** [comment]
> I think [#194](https://github.com/p0l0/hagruenbeck_cloud/issues/194) is related to the missing data on SE devices. Can you please check if with release 1.0.0 you get data for the entities?

Hi 

also no data with version 1.0.0

Aktueller Durchfluss
Unbekannt
Fehler
Problem
Regenerationsschritt
Unbekannt
Restkapazität %
Unbekannt
Restkapazität m³
Unbekannt
Salzverbrauch
Unbekannt
Salzvorrat
Unbekannt
Verbleibende Menge / Zeit der aktuellen Regenerationsstufe
Unbekannt
Weichwasserverbrauch
Unbekannt

**Comment by RehaagJ** [comment]
Same here. I also thought that #194 was the culprit, but that seems to be solved now, the crash during initialization no longer happens. Still , most values are "unknown". (SE21)

**Comment by p0l0** [comment]
Can you please download the diagnostic from the integration and post it here? The diagnostic should be anonymous.

That would help me to find out why it is still not working.

**Comment by RehaagJ** [comment]
Sure! Thanks for looking into this!

[config_entry-gruenbeck_cloud-01JWR295YVXTBATEXPX7NXZA29.json](https://github.com/user-attachments/files/20578488/config_entry-gruenbeck_cloud-01JWR295YVXTBATEXPX7NXZA29.json)

**Comment by oliof** [comment]
The good news is that v1.0.1 does not seem to cause the rate limit to kick in which rendered the app in a non working state.

Looking at the device in HA it also looks like more entities are setup almost right, albeit "Unknown" in most cases.

I tried to see whether "current flow rate" would update. It did in realtime in the gruenbeck app but kept "Unknown" in HA.

I have attached a diagnostics file.

[config_entry-gruenbeck_cloud-01JX0WV6TGKNWFJQ02VKNWKZ8Y.json](https://github.com/user-attachments/files/20617816/config_entry-gruenbeck_cloud-01JX0WV6TGKNWFJQ02VKNWKZ8Y.json)

**Comment by oliof** [comment]
[config_entry-gruenbeck_cloud-01JX0Z89NBWAJJBJ5PQXH79BRT-1.json](https://github.com/user-attachments/files/20622654/config_entry-gruenbeck_cloud-01JX0Z89NBWAJJBJ5PQXH79BRT-1.json) as promised here is the diagnostic after 8 hours.

**Comment by DeivisIMIQ** [comment]
Hello !!, I'm facing the same Issue (Tried to remove re-install including the last version 1.0.1., without success... :/ I did some tests using Postman (API GET POST) and seems that the response of the Object has another format as expected by the Integration... (Sorry, Python is not my best Prog. Language 😅 and I can be Wrong), But I could not find the matching keys in the integration (translations/EN.json) and in the response Object/JSON from my Grünbeck Device  model... 
Perhaps the data I've got can help to the solution (?!?!?)...  (If not, just to know that I am also looking forward to a solution 😁 and crossing my fingers, because this integration ist Top !!!)
**Device:**
{'errors': [], 'timeZone': '+00:00', 'regMode': 0, 'estatus': 0, 'regTrig': '0', 'startUp': '2025.02.12', 'swVersion': '0004.0010', 'hwVersionCl': '00000006', 'sncu': 'AH2407230154', 'calcRegMo1': '04:15', 'installedOn': '2024-08-22T09:13:53.9897079', 'regMo1': '2025-06-09T02:00:00', 'type': 118, 'hasError': False, 'isRegistered': False, 'id': 'softliQ.SE/BS11002192', 'series': 'softliQ.SE', 'serialNumber': 'BS11002192', 'name': 'Zuhause', 'register': False}
**Configuration**
{'$pending': [], 'pallowemail': True, 'pallowpushnotification': True, 'phdist': 24, 'pmaxvolmaxcap': 1.46, 'psetsoft': 6, 'pminvolmaxcap': 0.54, 'pwashingout': 4.0, 'pminvolmincap': 0.125, 'pregtu3': '--:--', 'pcurrent': 700, 'pregmo2': '--:--', 'pregwe1': '--:--', 'planguage': 2, 'phcurrent': 300, 'psetcapmo': 20.0, 'ptimezone': '02:00', 'pmaxvolmincap': 0.35, 'pregfr1': '--:--', 'pmaintint': 365, 'pregmo1': '02:00', 'pfreqregvalve': 140, 'pregwe2': '--:--', 'poverload': False, 'phload': 400, 'pdate': '2025.06.09', 'pforcedregdist': 3, 'pmonregmeter': 20.0, 'pmonsalting': 75.0, 'pregtu2': '--:--', 'pregwe3': '--:--', 'pregsa1': '--:--', 'pbuzzer': True, 'ppratesoftwater': 0.0043, 'pregsu1': '--:--', 'pregth1': '--:--', 'pregtu1': '--:--', 'pinspect': 180, 'pmondisinf': True, 'pdstdby': True, 'pmode': 2, 'pregmode': False, 'pclearcntreg': False, 'purlcloud1': 'gruenbeckeu01.azure-devices.net', 'phunit': False, 'pregsa2': '--:--', 'pled': 8, 'pdlstauto': True, 'prawhard': 22, 'purlcloud2': '', 'phwashingout': 5.0, 'phbackwash': 12.0, 'pregsa3': '--:--', 'pregsu3': '--:--', 'pactcurrent': 3, 'pfreqblendvalve': 300, 'pnomflow': 1.8, 'pregsu2': '--:--', 'pmonblend': False, 'pregfr3': '--:--', 'pregmo3': '--:--', 'pbackwash': 9.0, 'purlcert': 'prodeugruenbeckfirmware.blob.core.windows.net/cert', 'pdurcurrent': 600, 'pledbright': 80, 'pload': 1400, 'pregfr2': '--:--', 'pmodedesinf': 1, 'pntpsync': True, 'pprateregwater': 0.0039, 'pfreqregvalve2': 200, 'pprateblending': 0.004, 'pclearerrmem': False, 'pclearcntwater': False, 'pcontinent': 0, 'pmaxdurdisinfect': 4.0, 'pregth3': '--:--', 'prinsing': 9.5, 'pvolhyg': 6.8, 'pregth2': '--:--'}
**Data**
{'isncu': None, 'iothub-enqueuedtime': '0001-01-01T00:00:00', 'mregstatus': 0, 'mregpercent1': 63, 'mregpercent2': 0, 'mresidcap1': 74, 'mresidcap2': 244, 'mrescapa1': 281.0, 'mrescapa2': 933.0, 'mflow1': 0.0, 'mflow2': 0.0, 'mflowblend': 0.0, 'mstep2': False, 'mreswatadmod': 0.0, 'msaltrange': True, 'mtemp': -9.0, 'mcountreg': 73, 'mcountwater1': 21, 'mcountwater2': 0, 'mcountwatertank': 36, 'msaltusage': 15.0, 'mlifeadsorb': 0, 'mhardsoftw': 5.0, 'mcapacity': 6.0, 'maverage': 181, 'mstddev': 40, 'mflowmax': 1.76, 'mflowmax1reg2': 0.0, 'mflowmax2reg1': 0.0, 'mendreg1': '04:36', 'mendreg2': '00:00', 'mlime': 5, 'minspect': 0, 'mcountregservice': 0, 'hwaterusage0': 72, 'hsaltusage0': 101, 'hlime0': 5822}

**Comment by Andergast** [comment]
I can also confirm rate limiting in the iPhone app after enabling the integration and non-updating entities in HA - any more logs needed? I also own a SE21

**Comment by moerpel** [comment]
I think we are coming closer to a solution.
To me it looks like the data is being delivered by the API. Just an operand error follows.

2025-06-29 10:36:01.447 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Requesting URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1100xxxx/realtime/refresh?api-version=2024-05-02 with method POST
2025-06-29 10:36:01.536 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Response from URL https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1100xxxx/realtime/refresh?api-version=2024-05-02 with status 200 was {'isncu': None, 'iothub-enqueuedtime': '0001-01-01T00:00:00', 'mregstatus': 0, 'mregpercent1': 100, 'mregpercent2': 0, 'mresidcap1': 95, 'mresidcap2': 233, 'mrescapa1': 520.0, 'mrescapa2': 1272.0, 'mflow1': 0.0, 'mflow2': 0.0, 'mflowblend': 0.0, 'mstep2': False, 'mreswatadmod': 0.0, 'msaltrange': True, 'mtemp': -9.0, 'mcountreg': 165, 'mcountwater1': 51, 'mcountwater2': 0, 'mcountwatertank': 73, 'msaltusage': 30.0, 'mlifeadsorb': 0, 'mhardsoftw': 5.0, 'mcapacity': 6.0, 'maverage': 169, 'mstddev': 17, 'mflowmax': 1.78, 'mflowmax1reg2': 0.0, 'mflowmax2reg1': 0.0, 'mendreg1': '04:54', 'mendreg2': '00:00', 'mlime': 10, 'minspect': 55, 'mcountregservice': 0, 'hwaterusage0': 25, 'hsaltusage0': 150, 'hlime0': 5}
2025-06-29 10:36:01.536 ERROR (MainThread) [custom_components.gruenbeck_cloud.coordinator] Error fetching gruenbeck_cloud data: Unable to get data from API: unsupported operand type(s) for |: 'dict' and 'str'
2025-06-29 10:36:01.536 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Finished fetching gruenbeck_cloud data in 0.540 seconds (success: False)

**Comment by p0l0** [comment]
Can you please test if with version 1.0.5 the error with the response from the refresh endpoint is fixed?

**Comment by moerpel** [comment]
That looks way better than before:

<img width="332" height="599" alt="Image" src="https://github.com/user-attachments/assets/99c2ae7a-dea5-465e-a977-7ae0d7501394" />

2025-07-13 15:15:37.189 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:15:52.184 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:15:52.184 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:16:07.183 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:16:07.183 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:16:22.184 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:16:22.184 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:16:38.180 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:16:38.180 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:16:53.180 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:16:53.181 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:17:09.185 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:17:09.185 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:17:24.182 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:17:24.182 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data
2025-07-13 15:17:40.180 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] WebSocket Message received: {"type":6}
2025-07-13 15:17:40.181 DEBUG (MainThread) [custom_components.gruenbeck_cloud.coordinator] Manually updated gruenbeck_cloud data

**Comment by Friedi1970** [comment]
Thanks for fixing login resp. logout problem!!! Important step of improvement 🙏

I think the sensor values given with m³ can't be correct. I mean a capacity of more than 1000m³ isn't possible, right?

<img width="844" height="354" alt="Image" src="https://github.com/user-attachments/assets/c6694629-e252-40cf-95b2-15e49fc7b4ad" />

Actually, I have the same weird values and also some other sensor values seem to be not correct. Do you have an idea about that problem.

**Comment by dellimo** [comment]
I have testet 1.0.5 several times. It behaves all the time like this - After first login all values are unknown. A few minutes later it showing up the first values, but all are frozen. No updates anymore. At this time the grünbeck app is frozen, too. If I disable the grünbeck cloud, the grünbeck app is again running. Very strange. 

BTW - the unit of "Restkapazität" has to changed in L (Liter).  :-)

**Comment by Friedi1970** [comment]
> BTW - the unit of "Restkapazität" has to changed in L (Liter). :-)

Yep, and the "Weichwasserverbrauch" from L to m³ :-)

**Comment by makerwolf** [comment]
I just got my SE21 installed and immediately added your integration. I got values once, but it seems it is now stuck. What could i do to troubleshoot this issue?

actually I'm facing exactly the same issue as dellimo in his last comment. Grünbeck app is frozen too.

**Comment by MichelR90** [comment]
I have the same issue like makerwolf! Is there any solution yet? Is there a ban on side of Grünbeck? 
The values are frozen as described but the connectivity itself is not! I can still change settings etc. and yes you can prove it. LED-Ring blinks and boost mode can be activated but the values are not actualizied

**Comment by makerwolf** [comment]
There is a ban on side of Grünbeck. I've tried my best to contact the company and offer help creating local api access, just like BWT has, but the company has gone silent. 

It may be helpful if you also contact Grünbeck support and kindly but firmly ask for local api access to our own data. They're quite friendly, also supportive, but unfortunately only support.

**Comment by MichelR90** [comment]
I'm in contact to Grünbeck, but I think they right now don't want to work on a local api access. I also couldn't figure out why the access is blocked after the first connection. Anyone has an Idea what is happening here?

**Comment by sventiffe** [comment]
I've reached out to them as well and they made it pretty clear that they are not planning to support integration into smart home systems such as HA.

**Comment by MichelR90** [comment]
Yes that's right. So we have to find a solution on our own.

**Comment by sventiffe** [comment]
Wondering if monitoring network traffic from the device might help in logging the data at least. I wonder if I could set up a Wifi access point on my RasPi via https://community.home-assistant.io/t/add-on-create-a-wi-fi-access-point-with-hass-io-access-point/321732 for example, connect the device to this custom Wifi and route the traffic through the Raspi to the Gruenbeck cloud.

**Comment by MichelR90** [comment]
I did some research and lookup what is happening. I'm pretty sure the problem is that the api is not disconnected. (you can verify if the integration is not startet yet and you let the original app open for more then 10 minutes it will freeze) so let us implement two things. The API should disconnect after at least 5 minutes and the reconnect should not be more often than every 30 minutes. 
Yes I know live update would be cool but for now I'm sure it will not work...
My programming skills are not very good so i hope on the community. But never the less I will give it a try and tell you if solution is found.

**Comment by makerwolf** [comment]
Honestly, hearing they don't want to make local api access available makes me want to return my unit and swap it for a BWT. 

It's not worth trying to access their cloud API, I want realtime local data without having to rely on their cloud service. Especially if all the data we can get is crippled to what their app delivers.

**Comment by MichelR90** [comment]
We for sure all want this function but as long as they don't want we wil not get this. 
So I would be happy if we have a workaround as described before

**Comment by Bretzi** [comment]
Hi, I have a Grünbeck SE 18. What I've discovered is that there are significantly fewer sensor points and settings. Real-time data for the day is also unavailable. Water consumption, salt consumption, and limescale are retrieved via JSON for yesterday, month, and year. Basic system information, such as the serial number, is accessed via the interface. When I manually enter my login credentials using the previously retrieved bearer, I was able to populate the sensors once via JSON. However, after a certain time, the login no longer works. Perhaps a daily login would be sufficient to retrieve the previous day's data. Here is the JSON data:

https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1234567?api-version=2024-05-02
{
  "errors": [],
  "timeZone": "+00:00",
  "regMode": 0,
  "estatus": 0,
  "regTrig": "0",
  "startUp": "2025.09.25",
  "swVersion": "0004.0050",
  "hwVersionCl": "00000006",
  "sncu": "AH123456789",
  "calcRegMo1": "18:00",
  "installedOn": "2024-10-28T21:13:87.5792842",
  "regMo1": "2025-11-03T02:00:00",
  "type": 118,
  "hasError": false,
  "isRegistered": false,
  "id": "softliQ.SE/BS1234567",
  "series": "softliQ.SE",
  "serialNumber": "BS1234567",
  "name": "Gruenbeck",
  "register": false
}

https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1234567/measurements/history-values?api-version=2024-05-02

{
  "values": {
    "days": {
      "2025-10-27T00:00:00": {
        "saltUsage": 0,
        "waterUsage": 96,
        "limeRetained": 14
      },
      "2025-10-28T00:00:00": {
        "saltUsage": 151,
        "waterUsage": 54,
        "limeRetained": 9
      },
      "2025-10-29T00:00:00": {
        "saltUsage": 0,
        "waterUsage": 64,
        "limeRetained": 9
      },
      "2025-10-30T00:00:00": {
        "saltUsage": 0,
        "waterUsage": 102,
        "limeRetained": 17
      },
      "2025-10-31T00:00:00": {
        "saltUsage": 0,
        "waterUsage": 114,
        "limeRetained": 17
      },
      "2025-11-01T00:00:00": {
        "saltUsage": 151,
        "waterUsage": 60,
        "limeRetained": 9
      },
      "2025-11-02T00:00:00": {
        "saltUsage": 0,
        "waterUsage": 45,
        "limeRetained": 8
      }
    },
    "months": {
      "2025-09-01T00:00:00": {
        "saltUsage": 3156,
        "waterUsage": 1539,
        "limeRetained": 16111
      },
      "2025-10-01T00:00:00": {
        "saltUsage": 1118,
        "waterUsage": 2338,
        "limeRetained": 362
      },
      "2025-11-01T00:00:00": {
        "saltUsage": 151,
        "waterUsage": 105,
        "limeRetained": 17
      }
    },
    "years": {
      "2025-01-01T00:00:00": {
        "saltUsage": 4425,
        "waterUsage": 3982,
        "limeRetained": 16490
      }
    }
  }
}

**Comment by sventiffe** [comment]
In a similar way, also other endpoints seem to retrieve the correct information for my device:
https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1234567/parameters
https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1234567/measurements/salt
https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BS1234567/measurements/water

What others also observed already is that upon installation, quite some values are retrieved (and your examples retrieve the history as shown in the app). @p0l0 Would it be possible to add a functionality that repeats the login on a regular base and retrieves the current data?

Is your extension also able to store the historic data?

While this would not provide realtime support for many parameters, it would offer an integrated monitoring in Home Assistant on par with the mobile app and potentially allow some automation (around error reporting or low salt warning, for example).

**Comment by nsowen** [comment]
First, thanks for the implementation! What I noticed when analyzing the Grünbeck App traffic, is that there is no Websocket connection opened for the SE series. In exchange, the app polls the state every 60 seconds. Well this seems like an odd decision by the vendor to switch to frequent polling for such a usecase.

But there is more calls, which makes it even more odd, because there are repetitive init and exit calls, anyway here is what's happening:

1. Initiates the update call
POST https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BSxxxxxxxxxx/realtime/refresh?api-version=2024-05-02 => 200 OK

2. Enters update call
POST https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BSxxxxxxxxxx/realtime/enter?api-version=2024-05-02 => 200 OK

3. Performs update call every 60 seconds
GET https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BSxxxxxxxxxx/update?api-version=2024-05-02
`{"mrescapa1":1056.0,"mRescapa2":1428.0,"mflow1":0.0,"mflow2":0.0,"mflowblend":0.0,"msaltrange":true,"mregstatus":0,"mflowreg2":0.0,"mregpercent2":0,"mresidcap1":96,"mresidcap2":130,"mstep2":false,"mreswatadmod":0.0,"iregtrig":"0","mcountwater1":23,"mlime":5,"mtemp":-9.0,"type":121,"hasError":false,"isRegistered":false,"mcountreg":41,"mcapacity":15.4,"msaltusage":16.0,"mcountwatertank":42,"id":"softliQ.SE/BSxxxxxxxxxx","series":"softliQ.SE","serialNumber":"BSxxxxxxxxxx","name":"xxx","register":false}`

4. When quitting this realtime update
POST https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BSxxxxxxxxxx/realtime/leave?api-version=2024-05-02 => 200 OK
followed by

5. POST https://prod-eu-gruenbeck-api.azurewebsites.net/api/devices/softliQ.SE/BSxxxxxxxxxx/realtime/off?api-version=2024-05-02 => 200 OK

I would suggest is:
a) disable WS connections for SE devices - there is no data ever received anyway, right?
b) perform the realtime update initiation once, do the update call once a minute. When existing, call leave + off calls. 

I bet this will keep the session alive for the SE series, haven't tried it yet.

**Comment by MichelR90** [comment]
@nsowen your did the right research! This is the solution if somenone can implememt it, im very sure it will work!

**Comment by nsowen** [comment]
@MichelR90 Ok, I tried to get this working - even though I'm not really a python guy - but while debugging the iPhone App the server updates stopped, so I guess there is some kind of bug or in the server, or the server stops updating when we don't call realtime/leave + realime/off. If you are interested, here are my branches, both "work in progress". 

https://github.com/nsowen/pygruenbeck_cloud/tree/feature/se-series (see modified example.py)
https://github.com/nsowen/hagruenbeck_cloud/tree/feature/se-series

Update frequency is in fact once per second, so I assume this is good enough for "almost realtime".

**Comment by makerwolf** [comment]
I am trying to install your repo via hacs, but since you did not make a release on your branch I'm failing to get it installed for testing. I guess I'll try a manual install later ...

**Comment by makerwolf** [comment]
It seems it works better, I did get some current data. But unfortunately it stopped updating briefly after. If I open the grübeck app it starts working again briefly, but then stops updating again :(

**Comment by nsowen** [comment]
I did not get behind that black magic either. I’m afraid I missed something

**Comment by sventiffe** [comment]
I have a working solution with this script: https://gist.github.com/sventiffe/52e5c781ce44c7c76bcb2a5740f13aed

The script polls the data once per second following nsowen's sequence of endpoint calls. Instead of entering it and running updates I've decided to go through all five calls once per second. When running it, I do get live updates both in the script and in the app. Is this of any help?

**Comment by samyness** [comment]
Thanks for the research — I think this gives us a clear path for softliQ:SE* support in hagruenbeck_cloud. 

I think for the SE series the Grünbeck app does not open a WebSocket at all, and instead relies on REST polling via /update with the “realtime” init/exit calls.

**Proposed implementation for SE devices (softliQ:SE):**

1. Detect SE devices and switch to a dedicated SE mode
After the device is selected (e.g. after set_device_from_id() / get_device_infos()), check device.series (starts with softliQ.SE) and store a flag like is_se_device = True so we can use a different update strategy than SD/SC.

2. Disable WebSocket for SE devices
For softliQ.SE*, do not start the WebSocket listener (connect()/listen()), because the app traffic indicates there is no WS for SE and no data is received via WS anyway. This also avoids keeping unnecessary connections open.

3. Implement SE updates via REST polling following the app flow (realtime/* + /update)
Use the same sequence as the app:
- POST /realtime/refresh
- POST /realtime/enter
- then poll GET /update periodically (e.g. every 60 seconds like the app; optionally configurable)
- on shutdown/exit, always call POST /realtime/leave + POST /realtime/off to clean up

**Fallback / robustness option:** If the backend tends to stop updating unless we “close” properly (or if sessions get weird), we can use the brute-force approach and run all five calls per poll: refresh → enter → update → leave → off (some tests suggest this is more stable).

I’ll try to fork the repository soon and experiment with implementing this SE-specific update path. 
If I get something stable, I’ll share a branch/PR for testing.

**Comment by nsowen** [comment]
Before doing all that, please check my fork first, because it does all that already. Anyway, it stops working after some time. 

https://github.com/p0l0/hagruenbeck_cloud/issues/117#issuecomment-3719675747

**Comment by MichelR90** [comment]
@nsowenDo you how long it takes until it stops?  I will test your fork. I'm pretty sure it is 10 minutes. The original app stops in the same range until closed and newly connected. If you poll to often I believe the ban your ip...
-------- Ursprüngliche Nachricht --------Von: Nils Sowen ***@***.***> Datum: 14.01.26  14:41  (GMT+01:00) An: p0l0/hagruenbeck_cloud ***@***.***> Cc: MichelR90 ***@***.***>, Mention ***@***.***> Betreff: Re: [p0l0/hagruenbeck_cloud] Add support for SE Series (Issue #117) nsowen left a comment (p0l0/hagruenbeck_cloud#117)
Before doing all that, please check my fork first, because it does all that already. Anyway, it stops working after some time.

**Comment by nsowen** [comment]
Ah ok maybe I did not wait long enough during capturing the traffic from the original app. I will check this out. Poll frequency is once per second on the original app, so this is not the issue, there is some other rate limiting in place

**Comment by sventiffe** [comment]
Can you clarify what "some time" means? My script has been running now for 30 minutes and both my computer as well as my app receive real time updates.

**Comment by nsowen** [comment]
@sventiffe could be around 10 minutes. I understand your script is working, but that’s not what the app is doing: it will not do the full request cycle every second, but call update only, but only for a certain amount of time- which still has to be anaylzed. Side note - just my 2 cents: I wouldn’t risk increasing the call count if not really needed, because this could actually cause additional load and trouble on their service, and in turn invite Grünbeck to restrict even more? One should try to simulate exactly what their app is doing to avoid any operational issues.

**Comment by sventiffe** [comment]
@nsowen I do not think your premise is correct: we are not emulating the app to begin with. If you want to simulate exactly what their app is doing then you'd need to simulate the user journey the app and API has been designed for: occasional lookup of real time data. Any implementation for a 24/7 monitoring will hence significantly deviate from the design and purpose of the app.

I could alter my script to only poll the update endpoint without the five-step roundtrip. It would be interesting to see whether the updates will stop. From API design it might make sense because why would someone intentionally leave their app open for more than few minutes.

In terms of load: if we had a working solution it would cause 1QPS permanently for every Home Assistant user who integrates Gruenbeck SE devices. How large might this user group be? Few hundred users? The integration has 32 downloads according to the HACS overview (I might be wrong though). If that would make Grünbeck restrict their service... well. 

If we do the roundtrip, we'd approximately 5x the total QPS. Again, if we are speaking about few hundred users it would probably not move the needle much. I am not familiar with any QPS restrictions on Azure though. 

Another (crazy) idea: if the WLAN connection of the Gruenbeck device would be routed through Home Assistant -- for example by using HA as dedicated Wifi Access Point only for the device -- could we simply monitor the outgoing traffic of the device? Because that would be non invasive to begin with.

**Comment by samyness** [comment]
> if the WLAN connection of the Gruenbeck device would be routed through Home Assistant -- for example by using HA as dedicated Wifi Access Point only for the device -- could we simply monitor the outgoing traffic of the device? Because that would be non invasive to begin with.

Yes, that’s a reasonable non-invasive first step — but it has limits.

If you route the unit through a dedicated AP (HA/RasPi/OpenWrt), you can definitely capture metadata (which endpoints it talks to, request frequency, reconnect behavior, timeouts, etc.). That’s great for diagnosing “it stops updating after X minutes” issues.

However, the actual payload is HTTPS/TLS, so you typically won’t see the JSON/content unless you do TLS interception (MITM) — which is usually not feasible on IoT devices (no custom CA install / certificate pinning) and is no longer non-invasive.

So: useful for understanding behavior and timing, but probably not enough to extract the sensor values themselves.

**Comment by nsowen** [comment]
@sventiffe You got me wrong here: I'm not saying that the API usage should be *exactly* like the app is using it, but one should aim for the same flow (1QPS), and not for "this somehow works but we don't exactly know why" (5QPS) - so this is just fair play.

**Comment by sventiffe** [comment]
@nsowen "this somehow works but we don't exactly know why" --> i disagree. It works (not somehow) and we know exactly why (because it uses an end user workflow as if they opened the app once per second), and it is extremely easy to implement (as you would not need to deal with a stateful process like "refresh the start/stop sequence every x minutes")

What we do not exactly know are things like "what happens to the server and client states if we opened the connection, polled updates for x time intervals but then the process died without the two required concluding endpoint calls". In my experiments, skipping these concluding calls resulted in the app not getting updates anymore either and requiring a restart. Similar to the current behavior of the integration. 

For supporting the SE series I think the primary problem to solve is to get a working solution. If overall QPS are something that you'd be concerned about you could limit the updates to once every 5 seconds which will bring you near to the 1 QPS again. Even then, a single user's traffic would increase by ~100.000x if you compare "24/7 1 QPS for monitoring" with "user opens the app once per day and looks at it for 30 seconds". So that would be from my perspective the primary validation to proof, whether Grünbeck would tolerate that. 

If that would be the case, I would assume that increasing 1 QPS to 5 QPS would be tolerable, in particular if we talk about a user base of O(10) users, or maybe O(100) users globally.

**Comment by nsowen** [comment]
@sventiffe Ok, this is simply my personal view, maybe I have been doing web ops for too long. Even for 100 active users we are talking about 100 op/s vs. 500 op/s on top of their overall traffic, which is quite huge, assuming the update frequency is not reduced like you proposed. IMHO both ways would be easy to implement.

**Comment by makerwolf** [comment]
Nice to see work progress here! Maybe it is even a good thing to load up their system, to make them aware that there is indeed large interest in local control of their products, and it would actually save them a lot of cost and bring them good PR by implementing it.

**Comment by sventiffe** [comment]
I don't want to get into academic discussions here, but just to outline my thinking (in particular to @nsowen who looks at it from a different perspective). The Android app apparently has 50k downloads, and according to our new ML overlords the iOS app might be in the tier of 10k to 50k installations based on the feedback rate. So let's assume we have 25k users on mobile and a corresponding 25k total installation base of Gruenbeck devices in their market. 

The actual load on their backends will not be dominated by the casual use of the mobile apps and the endpoints we are discussing here. My assumption is that the devices need to send updates to their backends every second as a push (I doubt they can pull data on demand). So that would be 25k QPS in this model on their backends only from the devices independently from users requesting data. Given that I did not try to investigate which endpoints the devices are sending their data to, I only assume it's also on Azure, so this traffic estimate would be the baseline for assessing traffic increase.

The other unknown is the popularity of this integration. HACS tells me that the current integration has been dowloaded 32 times with 83 stars on the project (which sounds odd ratio). But as this integration is focusing on monitoring as opposed to actively controlling your home, I do assume it is targeting a niche user base of HA geeks. Hence I believe my initial estimate of a low user count might be plausible. 

If we caused additional 500 QPS on top of 25k QPS I would argue from a backend developer perspective that this is close to noise. If we are in a one-digit-percent increase of traffic we should be well within what a backend should have as buffer. If we have really a lot of traffic as @makerwolf suggests, it could raise a signal to them about a potential market. 

All that being said, I am not the maintainer of this project and any of my base assumptions above might be totally wrong. I would just like to monitor my water consumption for the lolz and get a reliable warning when salt is low. :)

**Comment by nsowen** [comment]
@sventiffe That's what all of us want :-) https://alerts.home-assistant.io/alerts/homematicip_cloud/ -> that's what happened to eQ3 users, just an example on how things might turn out if the userbase grows, in that case the iobroker people didn't care. But you're right in the end, for the moment, small userbase, low risk. Let's get to work then :-)

**Comment by frabu-t8p** [comment]
Just catched up with this thread as I have the same dreaded problem of blocked App after HA Gruenbeck install.   SE 24 here.  As soon as a workaround is around, happy to play with it ;-)

**Comment by titro** [comment]
> Just catched up with this thread as I have the same dreaded problem of blocked App after HA Gruenbeck install. SE 24 here. As soon as a workaround is around, happy to play with it ;-)

Same here with a SE24.

Integration install is working and values are collected. But only once. After that no further Updates in HA or the grünbeck App.
After deleting the Integration in HA the App is working the other Day without adding the softliq again. So seems that blocking or quota is resetted after some time or daily.

**Comment by frabu-t8p** [comment]
Is anything happening here?

**Comment by flo-m** [comment]
no, and additionally grünbeck stopped their activities in implementing a mqtt / modbus / ... API.

I tried to build a custom API but sooner or later I get blocked and don't receive any updates.

**Comment by makerwolf** [comment]
Oh that really sucks to hear! How did you hear about this? I'll need to give them another call ...

**Comment by flo-m** [comment]
i asked and they answered (translated from german):
"Due to the current market situation and the associated resource constraints, we are unfortunately forced to postpone the project indefinitely.

At this time, we are unable to say whether or when we will resume further development activities regarding an open interface."

**Comment by makerwolf** [comment]
So in other words: not enough people are requesting this change. *sigh* 
thanks for the update flo!

**Comment by Friedi1970** [comment]
I do not think too less people requested but they are just not interested ...

<img width="1298" height="960" alt="Image" src="https://github.com/user-attachments/assets/3b285a5e-8eae-494e-8066-62fe9c3615d2" />

**Comment by flo-m** [comment]
makes perfectly sense to transfer your data to the microsoft azure cloud (instead of local data access) and tell us that this is because of their strong focus on security.

**Comment by nsowen** [comment]
Pushed the changes based on @sventiffe 's work - seems to be working on my end. Set the update interval to 10 seconds which should be more than enough.

Also took the opportunity to clean up the entity situation a bit: the SE series doesn't support a large chunk of the existing entities, but comes with a few of its own instead, so I've tried to reflect that properly rather than having a bunch of unavailable/unknown values showing up.

Would love to hear if this works for others too ... especially great to hear from SL/SD users if I broke anything.

1. https://github.com/nsowen/pygruenbeck_cloud/tree/feature/se-series
2. https://github.com/nsowen/hagruenbeck_cloud/tree/feature/se-series

Will create the Pull Request if I get a green light!

**Comment by frabu-t8p** [comment]
I am getting "Unexpected Error" after entering the credentials in HA after installation via HACS

EDIT:
I could successfully register my device on second attempt. But all Sensors are showing up as "Unknown".
Some more patience needed... Sensors are now slowly tickling through . 

I keep you posted how it goes. Thanks very much for your efforts!

What I noticed that units are still mixed up:
- Restkapazität should be liter (not m3).
- Weichwasserverbrauch should be m3 (not liter).

**Comment by frabu-t8p** [comment]
No. Stops updating after a short time and the App is blocked. Same symptoms as previously.

**Comment by nsowen** [comment]
Yep, also noticed that after some time. So I assume the Grünbeck service is actively blocking further updates when attempting to retrieve frequent updates :-(

**Comment by frabu-t8p** [comment]
That is exactly what they are doing.  Gruenbeck is actively preventing users to ingrate this into any smart home system.

**Comment by flo-m** [comment]
it's possible to deactivate the extension for 15min and reactivate, then the app and the HA data is updating correctly until it stops some time in the future.
Any idea how to automate such an hard disconnect / deactivate?

**Comment by MichelR90** [comment]
To make things a bit clearer. Gruenbeck cancels the connection after 10 minutes.  It's a hard limit. Believe me.  Proof it with the original app. After 10 minutes open you won't get new live data.  You close and reopen and here you are.  So to get it working a disconnect from the cloud at least every 10 minutes or much better after getting data would be the solution.If we than don't grab data in seconds but let's say every 10 to 20 minutes it will work I am sure

**Comment by nsowen** [comment]
@MichelR90 Ok - I will try that. I don't think there is a clear indicator that the data is stale, but I will just generate a hash on the response payload, and if it does not change for some time, we can disconnect and delay the next update attempts for a specific time.

**Comment by schinken-tio** [comment]
Are there any news / can I help to get this working somehow?

**Comment by sventiffe** [comment]
I did not try out the updated integration yet but i cannot confirm that Grünbeck is canceling connections after 10 minutes. I've just had my script running and retrieved updates on a 15 seconds poll base for 30 minutes without neither the script nor the app stopping working. 

So it should be possible to create an integration which per update does the full API call per update:
```
steps = [
     ("POST", "/realtime/refresh"),
     ("POST", "/realtime/enter"),
     ("GET",  "/update"),
     ("POST", "/realtime/leave"),
     ("POST", "/realtime/off")
]
```
We discussed before that this is not what the app does, agreed. Just pointing out that it is working. If i find time to create an integration based on this i will share it.

**Comment by sventiffe** [comment]
After running more testing and network analysis, I came to the conclusion that the SE / SD series requires a different integration approach to avoid hitting cloud-side rate limits. These limits seem to be different from the observed 10 minute limit of continuous updates fwiw. 

I've created a dedicated custom integration specifically for this, which you can find here: https://github.com/sventiffe/gruenbeck_cloud_se

This integration is designed for low-frequency monitoring of slow-moving basic parameters:

- What it does: Allows you to track precise long-term water usage statistics (perfect for HA's Energy/Water dashboard) and set up active alerts if your softener goes into error mode or runs low on salt.
- How it polls: Fetches data on a safe, configurable interval between 10 and 60 minutes (defaulting to 10 minutes, which so far seems to be running stable).
- Limitations: You cannot actively control the device or poll sub-minute flow rates (fast sensors are disabled by default to keep dashboards clean).

If you have a  softliQ.SE  or  softliQ.SD  device, feel free to give it a try! If you have this integration installed read the installation notes, you might need to remove it as one library might not be compatible. The repo also contains summaries of my investigations for those who might be interested. Feedback is very welcome.

**Comment by frabu-t8p** [comment]
Thanks for your efforts. Successfully installed. I'll let you know whether I get continuous data.

**Comment by Andergast** [comment]
Same here, will get back asap

**Comment by frabu-t8p** [comment]
Looks good so far. For the first time ever a Grünbeck SE plugin is logging continuos data 👍👍

**Comment by titro** [comment]
> Looks good so far. For the first time ever a Grünbeck SE plugin is logging continuos data 👍👍

Same here. Yiiiha

**Comment by Andergast** [comment]
Same here - no hickups observed. Neither in HA nor in the Grünbeck app 👍

How wise it was to buy the model with ethernet access to be able to do stuff locally 🙄

**Comment by schinken-tio** [comment]
Also works great for me. Thank you very much @sventiffe !

**Comment by sventiffe** [comment]
Thank you! I will let it run for a while and eventually submit it to HACS. If you like the integration please leave me a star on the project page :)

@p0l0 if you prefer to integrate the functionality into this plugin feel free to suggest an eventual merge. I assume though that the concept of both integrations differs quite a bit so it seems to be ok to have two integrations available. I do refer to this project for credits and for other models in the documentation fwiw. 

Ideally, Grünbeck would be more willing to invest into a Cloud API or local API (that would even reduce their load and costs on the Cloud). The argument about security and stable updates is bollocks. The quality of their software is pretty bad to begin with. For example, if my device is stuck in a blocking dialog on screen that requires user confirmation it would neither send out updates nor send me email notifications (about depleted salt for example). The fact that they silently block updates without error codes instead of letting the user know that they ran into quota depletion is not good product excellence either. Updates on device i did not receive so far. And security for a monitoring solution, how does that even matter? At least on my device i would not know which settings to change actively -- so read access is all i need. 

I did leave a note in my documentation for people who shop for new water softeners that Grünbeck might be not their primary choice if they are interested in smart home integration because of these reasons.

**Comment by Friedi1970** [comment]
First: Thank you so much 🙏

I installed your integration and will keep an eye on it.

**Comment by zehndi77** [comment]
Thank you! It seems to work fine for me, except for the Flow parameter, which you mentioned would be high-frequency.
But already this is a great leap forward.

**Comment by sventiffe** [comment]
Let's move feedback on the SE integration to https://github.com/sventiffe/gruenbeck_cloud_se please so that we do not spam this bug. The flow parameter indeed is not useful at all in the new integration as monitoring flow only makes sense with realtime updates like the app or the original integration would do.

**Comment by Musok2** [comment]
For SE series support: the daily water/salt measurements currently fail to
parse because the API returns an extra `limeRetained` field per entry that
`DailyUsageEntry` rejects (default `Undefined.RAISE`).

I opened a PR in the library that fixes this:
p0l0/pygruenbeck_cloud#203

Tested against a real softliQ.SE account — with the fix all 69 daily entries
parse correctly.
